### PR TITLE
Tidy up namespace

### PR DESCRIPTION
**Pull request by ericpre**
To avoid having the module (`docstrings`, `exceptions`, `conftest`) when using autocompletion, etc.

### Progress of the PR
- [x] Make `docstrings` module private,
- [x] Move `exceptions` to `utils` folder,
- [x] Move `conftest.py` to `tests` folder,
- [n/a] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/rosettasciio/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [n/a] add tests,
- [x] ready for review.

### CODE REVIEW

**Comment by sem-geologist** [comment]
just why? Is this java project? While I understand that docstrings could be "private", I don't grasp any benefits in hiding exceptions. So if Someone will want to catch the rsciio specific expection he will need to import private model? This is so wrong on many levels...

File: upcoming_changes/132.maintenance.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Tidy up ``rsccio`` namespace: privitise ``docstings``, ``exceptions`` and move ``conftest.py`` to tests folder.
```
Comment (jlaehne) [inline upcoming_changes/132.maintenance.rst:1]:
```suggestion
Tidy up ``rsciio`` namespace: privatise ``docstrings``, ``exceptions`` and move ``conftest.py`` to tests folder
```

**Comment by ericpre** [comment]
> just why? Is this java project? While I understand that docstrings could be "private", I don't grasp any benefits in hiding exceptions. So if Someone will want to catch the rsciio-specific exception he will need to import private module? This is so wrong on many levels...

Alright @sem-geologist, agreed, it makes sense to keep `exceptions` public, it is now moved to the utils folder.

At some point, in another PR, we should review the content of the utils folder and define what makes to keep as public and document accordingly explicitly in the API reference. At the moment, there is nothing about it in the API reference.

File: upcoming_changes/132.maintenance.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Tidy up ``rsciio`` namespace: privatise ``docstings``, move ``conftest.py`` and ``exceptions`` to tests and utils folder, respectively
```
Comment (jlaehne) [inline upcoming_changes/132.maintenance.rst:1]:
```suggestion
Tidy up ``rsciio`` namespace: privatise common ``docstrings``, move ``conftest.py`` and ``exceptions`` to tests and utils folder, respectively
```

Comment (jlaehne) [reply upcoming_changes/132.maintenance.rst:1]: still stings and not strings :-)

**Review by jlaehne — APPROVED** [review]

**Comment by jlaehne** [comment]
@sem-geologist if you agree with the updated version of this PR, I think we could merge it

**Comment by sem-geologist** [comment]
@ericpre , yes its ok

**Comment by ericpre** [comment]
Thanks @sem-geologist, @jlaehne!

**Pull request merged**